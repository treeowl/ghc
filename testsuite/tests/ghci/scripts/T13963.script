:set -XPolyKinds -XDataKinds -XRankNTypes
import GHC.Exts (TYPE, RuntimeRep(BoxedRep), Levity(Lifted))
type Pair (a :: TYPE rep) (b :: TYPE rep') rep'' = forall (r :: TYPE rep''). (a -> b -> r)
:kind Pair
:kind Pair Int
:kind Pair Int Float
:kind Pair Int Float ('BoxedRep 'Lifted)

:set -fprint-explicit-runtime-reps
:kind Pair
:kind Pair Int
:kind Pair Int Float
:kind Pair Int Float ('BoxedRep 'Lifted)
