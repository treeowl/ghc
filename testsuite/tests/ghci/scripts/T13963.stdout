Pair :: * -> * -> RuntimeRep -> *
Pair Int :: * -> RuntimeRep -> *
Pair Int Float :: RuntimeRep -> *
Pair Int Float ('BoxedRep 'Lifted) :: *
Pair :: TYPE rep -> TYPE rep' -> RuntimeRep -> *
Pair Int :: * -> RuntimeRep -> *
Pair Int Float :: RuntimeRep -> *
Pair Int Float ('BoxedRep 'Lifted) :: *
